2 minor Mango test failures on Windows only

@tonysun83 @garrensmith any ideas on this one?

I've finally got the Mango test suite running on Windows. Here is the current output:

```
C:\relax\couchdb\src\mango>..\..\dev\run -n 1 --admin=testuser:testpass nosetests
[ * ] Setup environment ... ok
[ * ] Ensure CouchDB is built ... ok
[ * ] Prepare configuration files ... ok
[ * ] Start node node1 ... ok
[ * ] Check node at http://127.0.0.1:15984/ ... ok
[ * ] Running cluster setup ... ok
[ * ] Exec command nosetests ... ...............SSS...................................................................SSSSSSSSSSSSSSSSSSSSSSSSSS.................SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS.SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS...................................F.FSS.......SSSSSSSS...........
======================================================================
FAIL: test_quorum_json_index (15-execution-stats-test.ExecutionStatsTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\relax\couchdb\src\mango\test\15-execution-stats-test.py", line 39, in test_quorum_json_index
    self.assertGreater(resp["execution_stats"]["execution_time_ms"], 0)
AssertionError: 0.0 not greater than 0
-------------------- >> begin captured logging << --------------------
urllib3.connectionpool: DEBUG: http://127.0.0.1:15984 "POST /mango_test_9a3431add7ae4911b713cf9333377743/_find HTTP/1.1" 200 None
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: test_simple_json_index (15-execution-stats-test.ExecutionStatsTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\relax\couchdb\src\mango\test\15-execution-stats-test.py", line 26, in test_simple_json_index
    self.assertGreater(resp["execution_stats"]["execution_time_ms"], 0)
AssertionError: 0.0 not greater than 0
-------------------- >> begin captured logging << --------------------
urllib3.connectionpool: DEBUG: http://127.0.0.1:15984 "POST /mango_test_9a3431add7ae4911b713cf9333377743/_find HTTP/1.1" 200 None
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 284 tests in 90.943s

FAILED (SKIP=128, failures=2)
```

These two errors occur every time, easily repeatable, it's not a heisenbug. 

Smells like possibly a real Windows-specific bug to me, but not sure how to diagnose/help fix it. I don't just want to disable the test, and I'd like these all to pass cleanly before we release 2.3.0.

It could be as simple as this:

```
C:\relax\couchdb>erl
Eshell V8.3  (abort with ^G)
1> os:timestamp().
{1542,92419,103000}
2> os:timestamp().
{1542,92421,478000}
```

In other words, `os:timestamp()` on Windows appears to be accurate only to the millisecond, not the microsecond. Could these tests be running so fast that the delta is indeed `0.0`?

--- a/Makefile.win
+++ b/Makefile.win
@@ -119,6 +119,8 @@ check: all
 	@$(MAKE) -f Makefile.win test-cluster-without-quorum
 	@$(MAKE) -f Makefile.win eunit
 	@$(MAKE) -f Makefile.win javascript
+	@$(MAKE) -f Makefile.win mango-test
+#	@$(MAKE) -f Makefile.win elixir
 
 
 .PHONY: eunit

--- a/Makefile.win
+++ b/Makefile.win
@@ -125,15 +125,21 @@ check: all
 
 .PHONY: eunit
 # target: eunit - Run EUnit tests, use EUNIT_OPTS to provide custom options
+eunit: export ERL_AFLAGS = $(shell echo "-config rel/files/eunit.config")
+eunit: export BUILDDIR = $(shell echo %cd%)
 eunit: couch
-	@set ERL_AFLAGS="-config rel/files/eunit.config" && set BUILDDIR = $(shell echo %cd%) && $(REBAR) setup_eunit 2> nul
-	@set ERL_AFLAGS="-config rel/files/eunit.config" && set BUILDDIR = $(shell echo %cd%) && $(REBAR) -r eunit $(EUNIT_OPTS)
+	@$(REBAR) setup_eunit 2> nul
+	@$(REBAR) -r eunit $(EUNIT_OPTS)
 
+setup-eunit: export BUILDDIR = $(shell pwd)
+setup-eunit: export ERL_AFLAGS = $(shell echo "-config rel/files/eunit.config")
 setup-eunit:
-	@set ERL_AFLAGS="-config rel/files/eunit.config" && set BUILDDIR = $(shell echo %cd%) && $(REBAR) setup_eunit 2> nul
+	@$(REBAR) setup_eunit 2> nul
 
+just-eunit: export BUILDDIR = $(shell pwd)
+just-eunit: export ERL_AFLAGS = $(shell echo "-config rel/files/eunit.config")
 just-eunit:
-	@set ERL_AFLAGS="-config rel/files/eunit.config" && set BUILDDIR = $(shell echo %cd%) && $(REBAR) -r eunit $(EUNIT_OPTS)
+	@$(REBAR) -r eunit $(EUNIT_OPTS)
 
 
 .PHONY: elixir

--- a/make.cmd
+++ b/make.cmd
@@ -1,0 +1,3 @@
+@ECHO OFF
+
+make.exe -f Makefile.win %*

--- a/src/mango/test/15-execution-stats-test.py
+++ b/src/mango/test/15-execution-stats-test.py
@@ -12,6 +12,7 @@
 
 
 import mango
+import os
 import unittest
 
 class ExecutionStatsTests(mango.UserDocsTests):
@@ -23,7 +24,10 @@ class ExecutionStatsTests(mango.UserDocsTests):
         self.assertEqual(resp["execution_stats"]["total_docs_examined"], 3)
         self.assertEqual(resp["execution_stats"]["total_quorum_docs_examined"], 0)
         self.assertEqual(resp["execution_stats"]["results_returned"], 3)
-        self.assertGreater(resp["execution_stats"]["execution_time_ms"], 0)
+        # See https://github.com/apache/couchdb/issues/1732
+        # Erlang os:timestamp() only has ms accuracy on Windows!
+        if os.name != 'nt':
+            self.assertGreater(resp["execution_stats"]["execution_time_ms"], 0)
 
     def test_no_execution_stats(self):
         resp = self.db.find({"age": {"$lt": 35}}, return_raw=True, executionStats=False)
@@ -36,7 +40,10 @@ class ExecutionStatsTests(mango.UserDocsTests):
         self.assertEqual(resp["execution_stats"]["total_docs_examined"], 0)
         self.assertEqual(resp["execution_stats"]["total_quorum_docs_examined"], 3)
         self.assertEqual(resp["execution_stats"]["results_returned"], 3)
-        self.assertGreater(resp["execution_stats"]["execution_time_ms"], 0)
+        # See https://github.com/apache/couchdb/issues/1732
+        # Erlang os:timestamp() only has ms accuracy on Windows!
+        if os.name != 'nt':
+            self.assertGreater(resp["execution_stats"]["execution_time_ms"], 0)
 
     def test_results_returned_limit(self):
         resp = self.db.find({"age": {"$lt": 35}}, limit=2, return_raw=True, executionStats=True)

--- a/test/elixir/lib/couch.ex
+++ b/test/elixir/lib/couch.ex
@@ -47,7 +47,7 @@ defmodule Couch do
   """
 
   def process_url(url) do
-    "http://localhost:15984" <> url
+    "http://127.0.0.1:15984" <> url
   end
 
   def process_request_headers(headers, options) do

--- a/test/elixir/mix.exs
+++ b/test/elixir/mix.exs
@@ -23,7 +23,7 @@ defmodule Foo.Mixfile do
     [
       # {:dep_from_hexpm, "~> 0.3.0"},
       {:httpotion, "~> 3.0"},
-      {:jiffy, "~> 0.14.11"}
+      {:jiffy, "~> 0.15.2"}
       # {:dep_from_git, git: "https://github.com/elixir-lang/my_dep.git", tag: "0.1.0"},
     ]
   end

--- a/test/elixir/run
+++ b/test/elixir/run
@@ -1,4 +1,6 @@
 #!/bin/bash -e
 cd "$(dirname "$0")"
+mix local.hex --force
+mix local.rebar --force
 mix deps.get
 mix test --trace

--- a/test/elixir/run.cmd
+++ b/test/elixir/run.cmd
@@ -1,5 +1,7 @@
 @ECHO OFF
 
 cd %~dp0
+call mix local.hex --force
+call mix local.rebar --force
 call mix deps.get
 call mix test --trace

--- a/test/elixir/test/replication_test.exs
+++ b/test/elixir/test/replication_test.exs
@@ -10,9 +10,9 @@ defmodule ReplicationTest do
   @admin_account "adm:pass"
   @db_pairs_prefixes [
     {"local-to-local", "", ""},
-    {"remote-to-local", "http://localhost:15984/", ""},
-    {"local-to-remote", "", "http://localhost:15984/"},
-    {"remote-to-remote", "http://localhost:15984/", "http://localhost:15984/"}
+    {"remote-to-local", "http://127.0.0.1:15984/", ""},
+    {"local-to-remote", "", "http://127.0.0.1:15984/"},
+    {"remote-to-remote", "http://127.0.0.1:15984/", "http://127.0.0.1:15984/"}
   ]
 
   # This should probably go into `make elixir` like what
@@ -31,7 +31,7 @@ defmodule ReplicationTest do
 
   test "source database not found with host" do
     name = random_db_name()
-    url = "http://localhost:15984/" <> name <> "_src"
+    url = "http://127.0.0.1:15984/" <> name <> "_src"
     check_not_found(url, name <> "_tgt")
   end
 
@@ -53,7 +53,7 @@ defmodule ReplicationTest do
     doc = %{"_id" => "doc1"}
     [doc] = save_docs(src_db_name, [doc])
 
-    result = replicate(src_db_name, "http://localhost:15984/" <> tgt_db_name)
+    result = replicate(src_db_name, "http://127.0.0.1:15984/" <> tgt_db_name)
     assert result["ok"]
     assert is_list(result["history"])
     history = Enum.at(result["history"], 0)
@@ -73,7 +73,7 @@ defmodule ReplicationTest do
     })
     [doc] = save_docs(src_db_name, [doc])
 
-    result = replicate(src_db_name, "http://localhost:15984/" <> tgt_db_name)
+    result = replicate(src_db_name, "http://127.0.0.1:15984/" <> tgt_db_name)
 
     assert result["ok"]
     assert is_list(result["history"])
@@ -157,7 +157,7 @@ defmodule ReplicationTest do
 
     save_docs(src_db_name, make_docs(1..6))
 
-    repl_src = "http://localhost:15984/" <> src_db_name
+    repl_src = "http://127.0.0.1:15984/" <> src_db_name
     repl_body = %{"continuous" => true}
     result = replicate(repl_src, tgt_db_name, body: repl_body)
 

